### PR TITLE
Part of #12164: Use Coords for track_design_update_max_min_coordinates

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

File: src/openrct2/ride/TrackDesign.cpp [hunk 1]
```diff
@@ -855,14 +855,14 @@ static void track_design_add_selection_tile(int16_t x, int16_t y)
...
+static void track_design_update_max_min_coordinates(const CoordsXYZ coords)
 {
-    _trackPreviewMin.x = std::min(_trackPreviewMin.x, static_cast<int32_t>(x));
-    _trackPreviewMax.x = std::max(_trackPreviewMax.x, static_cast<int32_t>(x));
-    _trackPreviewMin.y = std::min(_trackPreviewMin.y, static_cast<int32_t>(y));
-    _trackPreviewMax.y = std::max(_trackPreviewMax.y, static_cast<int32_t>(y));
-    _trackPreviewMin.z = std::min(_trackPreviewMin.z, static_cast<int32_t>(z));
-    _trackPreviewMax.z = std::max(_trackPreviewMax.z, static_cast<int32_t>(z));
+    _trackPreviewMin = { std::min(_trackPreviewMin.x, static_cast<int32_t>(coords.x)),
+                         std::min(_trackPreviewMin.y, static_cast<int32_t>(coords.y)),
```
Comment (duncanspumpkin) [inline src/openrct2/ride/TrackDesign.cpp:861]: The static casts will no longer be required

File: src/openrct2/ride/TrackDesign.cpp [hunk 2]
```diff
@@ -855,14 +855,12 @@ static void track_design_add_selection_tile(int16_t x, int16_t y)
     gMapSelectionTiles.push_back(CoordsXY(x, y));
 }
 
-static void track_design_update_max_min_coordinates(int16_t x, int16_t y, int16_t z)
+static void track_design_update_max_min_coordinates(const CoordsXYZ coords)
```
Comment (Gymnasiast) [inline src/openrct2/ride/TrackDesign.cpp:858]: Have you made sure to check if this function is always properly called with big Z coordinates?

Comment (frutiemax) [reply src/openrct2/ride/TrackDesign.cpp:858]: Big Z coordinates?

Comment (duncanspumpkin) [reply src/openrct2/ride/TrackDesign.cpp:858]: I've checked and this one is fine.

Comment (tupaschoal) [reply src/openrct2/ride/TrackDesign.cpp:858]: I'll mark this as resolved then.

@frutiemax this PR explains what are big and small coords: https://github.com/OpenRCT2/OpenRCT2/pull/12192

Comment (tupaschoal) [inline src/openrct2/ride/TrackDesign.cpp:858]:
```suggestion
static void track_design_update_max_min_coordinates(const CoordsXYZ& coords)
```

File: src/openrct2/ride/TrackDesign.cpp [hunk 3]
```diff
@@ -1289,8 +1287,8 @@ static int32_t track_design_place_all_scenery(
             TileCoordsXY offsets = { scenery.x, scenery.y };
             tileCoords += offsets.Rotate(rotation);
 
-            CoordsXY mapCoord = tileCoords.ToCoordsXY();
-            track_design_update_max_min_coordinates(mapCoord.x, mapCoord.y, originZ);
+            CoordsXYZ mapCoord = { tileCoords.ToCoordsXY(), originZ };
```
Comment (tupaschoal) [inline src/openrct2/ride/TrackDesign.cpp:1290]:
```suggestion
            auto mapCoord = CoordsXYZ{ tileCoords.ToCoordsXY(), originZ };
```

File: src/openrct2/ride/TrackDesign.cpp [hunk 4]
```diff
@@ -1534,15 +1532,15 @@ static bool track_design_place_ride(TrackDesign* td6, int16_t x, int16_t y, int1
             trackType = TRACK_ELEM_MULTIDIM_INVERTED_90_DEG_UP_TO_FLAT_QUARTER_LOOP;
         }
 
-        track_design_update_max_min_coordinates(x, y, z);
+        track_design_update_max_min_coordinates(_trackPreviewOrigin);
```
Comment (tupaschoal) [inline src/openrct2/ride/TrackDesign.cpp:1535]: This is a for loop and `x`, `y` and `z` are changed at the end of the for, whereas `trackPreviewOrigin` are kept constant, so please use the former variables instead:
```suggestion
        track_design_update_max_min_coordinates({ x, y, z });
```

File: src/openrct2/ride/TrackDesign.cpp [hunk 5]
```diff
@@ -1688,7 +1686,7 @@ static bool track_design_place_ride(TrackDesign* td6, int16_t x, int16_t y, int1
         x = rotatedEntranceMapPos.x + _trackPreviewOrigin.x;
         y = rotatedEntranceMapPos.y + _trackPreviewOrigin.y;
 
-        track_design_update_max_min_coordinates(x, y, z);
+        track_design_update_max_min_coordinates({ rotatedEntranceMapPos + _trackPreviewOrigin, z });
```
Comment (tupaschoal) [inline src/openrct2/ride/TrackDesign.cpp:1689]: I know `x` and `y` will have the same value, but if for some reason someone changes them, with this proposal your code will also work, with the diff you proposed, we might forget the coupling and break this

```suggestion
        track_design_update_max_min_coordinates({ x, y, z });
```

**Pull request merged**